zng/docs/spec.md

```diff
@@ -511,15 +538,16 @@ An application-specific payload has the following form:
 Here, `<control code>` is a decimal integer in the range 6-127 and `<payload>`
 is any UTF-8 string with escaped newlines.
 
-### Type Grammar
+### 4.2 Type Grammar
 
 Given the above textual definitions and the underlying ZNG specification, a
 grammar describing the textual type encodings is:
 ```
 <stype> := bool | byte | int16 | uint16 | int32 | uint32 | int64 | uint64 | float64
-         | string | bytes | bstring | enum | ip | net | time | duration | null
+         | string | bytes | bstring | enum | ip | port | net | time | duration | null
          | <alias-name>
 
+XXX - I don't see ctype referenced anywhere else. Shouldn't it be?
```

Comment:
I'm not a pro at these pseudo-grammars, but just following the pattern of what I see in the ones in this spec, it seems like we only define something in `<brackets>` if it's actually referenced elsewhere, and this one isn't referenced elsewhere, so I want to check if something needs to be corrected here.

Reply:
Good catch. We could just add:

```
<type-string> := <ctype>
```

to say that a type definition can be any container type. The caveat is that this isn't currently matched by the tzng implementation, which requires that a type definition be a record type. 

This could be ok based on a similar precedent of the zng spec being broader than the current implementation. If however we want the grammar to match the implementation, then we could have `<type-string> := <rtype>` and 

```
<rtype> :=  record [ <columns> ]
               | record [ ]

<ctype> :=  array [ <stype> ]
          | union [ <stype-list> ]
          | set [ <stype> ]
          | <rtype>
```

Reply:
I pushed a commit to do the above, and also updates `<column>` to consist of both simple and container types. (having realized as I did it that my comment above was incomplete)

Reply:
Cool, thanks @henridf! While looking at your commit, I noticed for the first time some excess space characters that were in there from before, so I've pushed yet another commit to improve that.